src/analysis/natural_parameterization.lean

```diff
@@ -0,0 +1,227 @@
+/-
+Copyright (c) 2023 Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémi Bottinelli
+-/
+import measure_theory.measure.lebesgue
+import analysis.calculus.monotone
+import data.set.function
+import analysis.bounded_variation
+import tactic.swap_var
+/-!
+# Natural Parameterization
+
+This file defines the notion of linear and natural parameterization for a function `f : ℝ → E` with
+pseudo-emetric structure on `E` with respect to a set `s : set ℝ` and "speed" `l : ℝ≥0`, and shows
+that if `f` has locally bounded variation on `s`, it can be obtained (up to distance zero, on `s`),
+as a composite `φ ∘ (variation_on_from_to f s a)`, where `φ` is linearly parameterized and `a ∈ s`.
+
+## Main definitions
+
+* `is_linearly_parameterized_on_by f s l`, stating that the speed of `f` on `s` is `l`.
+* `is_naturally_parameterized_on f s`, stating that the speed of `f` on `s` is `1`.
```

Comment:
I think I prefer the names `is_constant_speed_on` and `is_unit_speed_on` because they are shorter and less ambiguous (naturally parameterized could mean many other things).